### PR TITLE
Feat: custom flow support

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of the effort is to add custom-flow support #550.

In this PR I added API client support for CRUD operations on custom flows.
Unit tests were added as well.

### CODE REVIEW

File: client/custom_flow.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	SshKeys              []TemplateSshKey `json:"sshKeys,omitempty"`
+	GithubInstallationId int              `json:"githubInstallationId,omitempty"`
+	BitbucketClientKey   string           `json:"bitbucketClientKey,omitempty"`
+	IsBitbucketServer    bool             `json:"isBitbucketServer"`
+	IsGitlabEnterprise   bool             `json:"isGitLabEnterprise"`
+	IsGithubEnterprise   bool             `json:"isGitHubEnterprise"`
+	IsGitLab             bool             `json:"isGitLab"`
+	IsAzureDevOps        bool             `json:"isAzureDevOps" tfschema:"is_azure_devops"`
+	IsTerragruntRunAll   bool             `json:"isTerragruntRunAll"`
+}
```
Comment (yaronya) [inline client/custom_flow.go:18]: Is it possible to add some composition with the already existing attribute of the template resource?

File: client/custom_flow.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+type CustomFlow struct {
```
Comment (yaronya) [inline client/custom_flow.go:20]: Would be great to DRY this a bit and add a shared `struct` between this one and `CustomFlowCreatePayload`

File: client/custom_flow.go (hunk 1 above)
Comment (TomerHeber) [reply client/custom_flow.go:18]: Yes! It's possible. It makes more sense to me.

Embedded composition in golang is explicit. See this for an explanation:
https://blog.dlow.me/programming/golang/struct-embedding/#golang-struct-embedding

This means that I will have to make changes in all the code locations where Template is initialized
E.g.:
```
	template := client.Template{
		Id:               "id0",
		Name:             "template",
...

```

It's not an issue - just additional (many) cosmetic changes in the PR.
Let me know if I should proceed. Another option is to apply it as a follow-up PR.

File: client/custom_flow.go (hunk 2 above)
Comment (TomerHeber) [reply client/custom_flow.go:20]: They vary from a "tagging" perspective. (json and tfschema).
Trying to merge may become cumbersome, and the benefit is small. It is currently consistent with the rest of the code.

File: client/custom_flow.go (hunk 1 above)
Comment (yaronya) [reply client/custom_flow.go:18]: Yeah it's probably not worth it...

**Review by yaronya — APPROVED** [review]

**Pull request merged**